src/image.rs

```diff
@@ -1,113 +1,27 @@
-use std::marker::PhantomData;
 use std::slice;
 
 use libavif_sys as sys;
 
-pub struct RgbPixels<'a> {
-    owned: bool,
-    inner: sys::avifRGBImage,
-
-    phantom: PhantomData<&'a [u8]>,
+pub struct AvifImage {
```

Comment:
I'm not clear what this `AvifImage` type is supposed to represent.

Maybe instead, it'd be nice to have an `enum Image { Rgb(...), Rgba(...), Yuv444(...) }` ?

Reply:
The idea was to implement reading and writing things like the EXIF and ICCP in the future, and [keeping the same structure libavif has](https://github.com/AOMediaCodec/libavif/blob/master/include/avif/avif.h#L298-L346) seems to be a good option